File: 1.IFC/Codes/C/MapaLogistico.c
/*
Autor: Carlos Antunis Bonfim da Silva Santos
Objetivo:
Computar o comprimento de uma circunferência com raio
informado pelo usuário
*/
#include <stdio.h>

int main() {
    const double dr = 0.0001, dx = 0.01;
    double X_mapa;
    //Arquivo dos dados
    FILE *Grafico_das_Bifurcacoes = fopen("Bifurcacoes.dat", "w");

    //Para cada parametro possível
    for (double r = 3.0; r < 4.0 ; r+=dr) {
        //Para cada condição inicial
        for (double X_0 = 0.1; X_0 < 1.0; X_0+=dx) {
            X_mapa = X_0;
            //Obter o comportamento assintótico do mapa
            for (int Iteracao = 0; Iteracao < 15000; Iteracao++)
                X_mapa = r * X_mapa * (1.0 - X_mapa);

            //Escrever o valor do mapa do mapa
            fprintf(Grafico_das_Bifurcacoes, "%lf %lf\n", r, X_mapa);
        }
    }

    //Fechar o arquivo de dados
    fclose(Grafico_das_Bifurcacoes);

    return 0;
}

File: 1.IFC/Codes/Python/MapaLogistico.py
#Autor: Carlos Antunis Bonfim da Silva Santos
#Objetivo:
#Computar o comprimento de uma circunferência com raio
#informado pelo usuário

CONST_dr = 0.0001
CONST_dx = 0.01

#Com o arquivo de dados aberto
with open("Bifurcacoes.dat","w") as Grafico_das_Bifurcacoes:
    #Para cada parametro possível
    r = 3.0
    while (r < 4.0):
        #Para cada condição inicial
        X_0 = 0.1
        while(X_0 < 1.0):
            X_mapa = X_0
            for Iteracao in range(0, 15000):
                X_mapa = r * X_mapa * (1.0 - X_mapa)

            #Escrever o valor do mapa do mapa
            Grafico_das_Bifurcacoes.write(r, X_mapa)

            X_0 = X_0 + CONST_dx
        r = r + CONST_dr